src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryHelper.scala

```diff
@@ -35,17 +35,24 @@ class TaskOpFactoryHelper(
     TaskOp.Launch(taskInfo, newTask, Some(oldTask), createOperations)
   }
 
+  /**
+    * Returns a set of operations to reserve ALL resources (cpu, mem, ports, disk, etc.) and then create persistent
+    * volumes against them as needed
+    */
   def reserveAndCreateVolumes(
     frameworkId: FrameworkId,
     newTask: TaskStateOp.Reserve,
     resources: Iterable[Mesos.Resource],
-    localVolumes: Iterable[LocalVolume],
+    localVolumes: Iterable[(Option[Mesos.Resource.DiskInfo.Source], LocalVolume)],
```

Comment:
1) in general, I dislike tuples in interfaces. I could lie with it here, but it's but too obvious at first glance why the first one is an option

2) please try to prevent using generated Mesos proto classes when possible. In this case, the lack of an enum value ~ root makes this an Option. It would be nicer to get rid of the option and use a dedicated Marathon type (why not the existing `DiskType`?)